### PR TITLE
Update station to 1.12.0

### PR DESCRIPTION
**Pull request by shoichiaizawa**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

File: Casks/station.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 cask 'station' do
-  version '1.11.2'
-  sha256 '5c0bf5e0689547cd3c89110ac4e5efa10b6a698ccd52306955127a758dc64ec0'
+  version '1.12.0'
+  sha256 '5bcb8834cd2737fe98a6b1569ac06dbd7482c5b7eb0c59fd88c17a874d1e0ede'
 
   # github.com/getstation/desktop-app-releases was verified as official when first introduced to the cask
-  url "https://github.com/getstation/desktop-app-releases/releases/download/#{version}/Station-#{version}-mac.zip"
+  url "https://github.com/getstation/desktop-app-releases/releases/download/#{version}/browserX-#{version}-mac.zip"
```
Comment (commitay) [inline Casks/station.rb]: Is `browserX` the same app? 

The releases page has a `station` dmg: https://github.com/getstation/desktop-app-releases/releases/download/1.12.0/Station-1.12.0.dmg

Comment (shoichiaizawa) [reply Casks/station.rb]: @commitay 

### TL;DR

I believe the `browserX` and the `.dmg` versions are identical.

### A long answer... I did a few things to confirm their identicalness

I was going to mention on that point in the PR message above (which I didn't in the end...); the download URL has been changed for some reason as you noticed.

Up to the previous version (`1.11.2`), the word `Station` was part of the download URL instead of the word `browserX`, but for the latest version (`1.12.0`).

Prior to performing the `cask-repair -e -i station` command to update the `Station` cask with the URL modification, I tried downloading [browserX-1.12.0-mac.zip](https://github.com/getstation/desktop-app-releases/releases/download/1.12.0/browserX-1.12.0-mac.zip) to see what's in it – there was `Station.app` (latest 1.12.0) inside – thus, I submitted this PR.

And now, just in case, I have double-checked what's inside both [browserX-1.12.0-mac.zip](https://github.com/getstation/desktop-app-releases/releases/download/1.12.0/browserX-1.12.0-mac.zip) and [Station-1.12.0.dmg](https://github.com/getstation/desktop-app-releases/releases/download/1.12.0/Station-1.12.0.dmg); they both contain the same `Station.app` (latest 1.12.0), with 343 MB of the app size (according to Finder.app). The only difference was their creation time – `browserX` was created on 2018-05-17, 17:00 and `Station.dmg` was created on 2018-05-18, 02:00. (Note that these times may be adjusted to my locale UTC+9).

And finally, I did triple-check if they are identical or not by checking their MD5 hashes; it turned out that they are identical – here is the result:

```sh
# Please note that I renamed the app names, one for `browserX` and
# and the other for `dmg`
$ md5 Station-browserX.app/Contents/MacOS/Station
MD5 (Station-browserX.app/Contents/MacOS/Station) = f113fe1fa5c53f2bc3ab9fe4f82060c9

$ md5 Station-dmg.app/Contents/MacOS/Station
MD5 (Station-dmg.app/Contents/MacOS/Station) = f113fe1fa5c53f2bc3ab9fe4f82060c9
```

I suppose whether updating the cask with [browserX-1.12.0-mac.zip](https://github.com/getstation/desktop-app-releases/releases/download/1.12.0/browserX-1.12.0-mac.zip) or [Station-1.12.0.dmg](https://github.com/getstation/desktop-app-releases/releases/download/1.12.0/Station-1.12.0.dmg), either way should work fine.

Comment (commitay) [reply Casks/station.rb]: Thanks for checking @shoichiaizawa! I've changed this to the dmg to match the homepage download.
